File: LEETCODE/HASHTABLE/53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        mxsum,csum=-float('inf'),0
        size=len(nums)
        if size==1:
            return nums[0]
        for i in range(size):
            if csum<0:
                csum=0
            csum+=nums[i]
            mxsum=max(mxsum,csum)
        
        return mxsum
        

File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        
        int sum=nums[0],maxsum=nums[0];

        // Taking the sum and compare the with the new sum(which exist from the point of getting low)

        for(int i=1;i<nums.length;++i)
        {
            if(nums[i]>sum && sum<0)
            {
                sum=nums[i];    
            }
            else
            {
                sum+=nums[i];
            }
            maxsum=Math.max(maxsum,sum);
            if(sum<0) sum=0;
        }

        return maxsum;
    }
}